lib/metadata_json_lint.rb

```diff
@@ -136,10 +136,26 @@ def validate_requirements!(requirements)
       if requirement['name'] == 'pe'
         warn :requirements, "The 'pe' requirement is no longer supported by the Forge."
       end
+
+      begin
+        VersionRequirement.new(requirement.fetch('version_requirement', ''))
+      rescue ArgumentError => e
+        # Raised when the version_requirement provided could not be parsed
+        error :dependencies, "Invalid 'version_requirement' field in metadata.json: #{e}"
+      end
+
+      validate_puppet_ver!(requirement)
     end
   end
   module_function :validate_requirements!
 
+  def validate_puppet_ver!(range)
+    min_ver = range['version_requirement'].gsub(/[><=~^]/, '').split(' ')[0]
+
+    error(:requirements, "#{min_ver} is no longer a supported version of Puppet. Please use a valid range between 4.10.0 and 6.0.0.") if min_ver =~ /^[2-4]\.[0-9]\.\d$/
```

Comment:
Module authors should be free to support older versions. I get some people don't want to support EOL versions, but in practice sometimes you have to. Making this required will mean some users will stop using this. We have branches that support older versions and often do cherry picks. It's nice to know the metadata is correct, but if this is required I'll just have to rip the validation from our CI. Please make this configurable.

Mentioning the upper bound is also a bad idea, especially since you don't check for it anyway.

I'd also use `VersionRequirement` instead of parsing it yourself.

Reply:
@dhollinger I think there is confusion between schema and data with regards to what is "current stable". I agree that we should target the "current stable" version of the schema. This is different from the data.

If you have an up to date and totally valid metadata.json and it supports an EOL version of puppet, that should not throw an error by default. That violates the nature of the tool, which is to ensure the metadata.json follows the schema, as well as the principal of least surprise.

Reply:
An old version of puppet, even an unsupported one, in `metadata.json` is just part of linting, there's no need for or implication that the testing framework to support that version. That's up to the user.

I suggest using `--strict-supported-versions` and a `if options[:strict_supported_versions]` guard around the error, similar to [this](https://github.com/voxpupuli/metadata-json-lint/blob/39d756d015ef6e6ff75f75af06e5e5c0c1cf3bb7/lib/metadata_json_lint.rb#L106).